interop-testing/src/main/java/io/grpc/testing/integration/XdsTestServer.java

```diff
@@ -0,0 +1,142 @@
+   */
+  public static void main(String[] args) throws Exception {
+    final XdsTestServer server = new XdsTestServer();
+    server.parseArgs(args);
+    Runtime.getRuntime()
+        .addShutdownHook(
+            new Thread() {
+              @Override
+              @SuppressWarnings("CatchAndPrintStackTrace")
+              public void run() {
+                try {
+                  System.out.println("Shutting down");
+                  server.stop();
+                } catch (Exception e) {
+                  e.printStackTrace();
+                }
+              }
+            });
+    server.start();
+    System.out.println("Server started on port " + server.port);
+    server.blockUntilShutdown();
+  }
+
+  private void parseArgs(String[] args) {
+    boolean usage = false;
+    for (String arg : args) {
+      if (!arg.startsWith("--")) {
+        System.err.println("All arguments must start with '--': " + arg);
+        usage = true;
+        break;
+      }
+      String[] parts = arg.substring(2).split("=", 2);
+      String key = parts[0];
+      if ("help".equals(key)) {
+        usage = true;
+        break;
+      }
+      if (parts.length != 2) {
+        System.err.println("All arguments must be of the form --arg=value");
+        usage = true;
+        break;
+      }
+      String value = parts[1];
+      if ("port".equals(key)) {
+        port = Integer.valueOf(value);
+      } else if ("server_id".equals(key)) {
+        serverId = value;
+      } else {
+        System.err.println("Unknown argument: " + key);
+        usage = true;
+        break;
+      }
+    }
+
+    if (usage) {
+      XdsTestServer s = new XdsTestServer();
+      System.err.println(
+          "Usage: [ARGS...]"
+              + "\n"
+              + "\n  --port=INT          listening port for server."
+              + "\n                      Default: "
+              + s.port
+              + "\n  --server_id=STRING  server ID for response."
+              + "\n                      Default: "
+              + s.serverId);
+      System.exit(1);
+    }
+  }
+
+  private void start() throws Exception {
+    server = NettyServerBuilder.forPort(port).addService(new TestServiceImpl()).build().start();
+  }
+
+  private void stop() throws Exception {
+    server.shutdownNow();
+    if (!server.awaitTermination(5, TimeUnit.SECONDS)) {
+      System.err.println("Timed out waiting for server shutdown");
+    }
+  }
+
+  private void blockUntilShutdown() throws InterruptedException {
+    if (server != null) {
+      server.awaitTermination();
+    }
+  }
+
+  private class TestServiceImpl extends TestServiceGrpc.TestServiceImplBase {
+    private final String host;
+
+    private TestServiceImpl() {
+      String tmpHost;
+      try {
+        tmpHost = InetAddress.getLocalHost().getHostName();
+      } catch (UnknownHostException e) {
+        tmpHost = "failed to get host " + e;
```

Comment:
Why not fail the entire test immediately? Or just log the error and let host name be empty.

Reply:
Done